nmagent/cachedclient.go

```diff
@@ -0,0 +1,178 @@
+package nmagent
+
+import (
```

Comment:
can you use "github.com/patrickmn/go-cache" ? This has been used by DNC RC. Check the code in nodenetworkconfigreconciler.go

Reply:
[A little copying is better than a little dependency](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=568s).
That dependency is 10x bigger than this whole cached-client implementation, and significantly obfuscates the behavior of the client, which IMO should be a very simple pull-through cached client.